std/range/primitives.d

```diff
@@ -161,17 +161,11 @@ Params:
 Returns:
     true if R is an InputRange, false if not
  */
-template isInputRange(R)
-{
-    enum bool isInputRange = is(typeof(
-    (inout int = 0)
-    {
-        R r = R.init;     // can define a range object
-        if (r.empty) {}   // can test for empty
-        r.popFront;       // can invoke popFront()
-        auto h = r.front; // can get the front of the range
-    }));
-}
+enum bool isInputRange(R) =
+    is(typeof((R r) => R.init))
+    && is(ReturnType!((R r) => r.empty) == bool)
```

Comment:
This constraint is weaker than before. (The old constraint checked for cast(bool)r.empty being valid.)